File: 백준/Bronze/11606. Egg Drop/Egg Drop.cc
#include<bits/stdc++.h>
using namespace std;

int n, k;
int s, e;
int m;
string t;

int main()
{
	cin>>n>>k;
	s=1, e=k;
	while(n--)
	{
		cin>>m>>t;
		if(t=="SAFE") s=max(s, m);
		else e=min(e, m);
	}
	cout<<s+1<<" "<<e-1;
}

File: 백준/Bronze/26513. Egg Drop/Egg Drop.py
while True:
	n, m=map(int, input().split())
	if n==m==0: break
	A, B=2, m-1
	for i in range(n):
		a, b=input().split()
		a=int(a)
		if b=="SAFE": A=max(A, a+1)
		else: B=min(B, a-1)
	print(A, B)